### PR TITLE
CentOS7 Package Versions

### PR DESCRIPTION
**Pull request by citibeth**
This PR adds the versions of packages that are provided with CentOS7.  Benefits:

 1. Packages specified as `buildable=False` work better if Spack knows about the version number.  Previously, I had to lie about which version of Python was provided by my system (`2.7.8` instead of the actual `2.7.5`, which Spack didn't know about).

 2. Users not on CentOS7 can replicate the CentOS7 experience.  Ooh fun...

This PR also converts to `AutotoolsPackage` and `CMakePackage` as appropriate.

### CODE REVIEW

File: var/spack/repos/builtin/packages/curl/package.py [hunk 1]
```diff
@@ -42,14 +42,14 @@ class Curl(Package):
...
-        configure('--prefix=%s' % prefix,
-                  '--with-zlib=%s' % spec['zlib'].prefix,
-                  '--with-ssl=%s' % spec['openssl'].prefix)
-
-        make()
-        make("install")
+    def configure_args(self):
+        spec = self.spec
+        return [
+            '--prefix=%s' % prefix,
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/curl/package.py]: Prefix already gets appended, so you can leave it out.

File: var/spack/repos/builtin/packages/xz/package.py [hunk 2]
```diff
@@ -34,7 +34,7 @@ class Xz(Package):
     url      = "http://tukaani.org/xz/xz-5.2.0.tar.bz2"
 
     version('5.2.0', '867cc8611760240ebf3440bd6e170bb9')
-    version('5.2.2', 'f90c9a0c8b259aee2234c4e0d7fd70af')
+    version('5.2.2', 'f90c9a0c8b259aee2234c4e0d7fd70af') # CentOS7
 
     def install(self, spec, prefix):
```
Comment (citibeth) [inline var/spack/repos/builtin/packages/xz/package.py]: Does anyone know how to properly convert this to `AutotoolsPackage`?  It has a test in it...

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/xz/package.py]: Just delete the whole `install` method. Those tests are already part of `AutotoolsPackage`.

**Comment by citibeth** [comment]
@adamjstewart Thanks, I think I'm done with this PR.  Things look good for now; I'll think about converting Qt to AutotoolsPackage on another PR.

**Comment by tgamblin** [comment]
Updated w/current `develop`.

**Comment by citibeth** [comment]
Would you like any rebasing or something for this on my part?

**Comment by tgamblin** [comment]
@citibeth: it was just a comment so I used the in-browser updater (which ends up looking like I merged `develop` into the PR).  The only conflict was between your comment and @adamjstewart's `1.1.0e`.  It seems like this is something `git` could be smarter about resolving, if it knew what was comment and what was code...

**Comment by adamjstewart** [comment]
Out of curiosity, what breaks when you try to use external packages if the version is not included in the `package.py`? We should probably fix that as well. I don't see any reason to prevent people from using other versions.

**Comment by citibeth** [comment]
I forget

File: var/spack/repos/builtin/packages/python/package.py [hunk 3]
```diff
@@ -36,7 +36,7 @@
 import spack.util.spack_json as sjson
 
 
-class Python(Package):
+class Python(AutotoolsPackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/python/package.py]: Can you remove this change? The package isn't setup to use `AutotoolsPackage`.

File: var/spack/repos/builtin/packages/libssh2/package.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Libssh2(CMakePackage):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libssh2/package.py]: Was this package supposed to be included in this PR? It looks like it's already been merged.

File: var/spack/repos/builtin/packages/libpng/package.py [hunk 5]
```diff
@@ -46,5 +46,7 @@ class Libpng(AutotoolsPackage):
 
     # Required for qt@3
     version('1.2.57', 'dfcda3603e29dcc11870c48f838ef75b')
+    # CentOS7
+    version('1.5.13', '9c5a584d4eb5fe40d0f1bc2090112c65')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libpng/package.py]: Can you put this version above `1.2.57`?

**Comment by citibeth** [comment]
OK... I rebased, and this should be ready now.

File: var/spack/repos/builtin/packages/openssl/package.py [hunk 6]
```diff
@@ -38,10 +38,13 @@ class Openssl(Package):
...
-    # Note: Version 1.0.2 is the "long-term support" version that will
-    # remain supported until 2019. We could thus make this version the
-    # preferred version, if we find that many packages cannot handle
-    # version 1.1.
+    # Note:
+    # 1. Version 1.0.2 is the "long-term support" version that will
+    #    remain supported until 2019. We could thus make this version the
+    #    preferred version, if we find that many packages cannot handle
+    #    version 1.1.
+    # 1. Many of these versions have known vulnerabilities.
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openssl/package.py]: This should be a 2.

**Comment by adamjstewart** [comment]
Can you rebase this again?

**Comment by adamjstewart** [comment]
Ping @citibeth. When this is rebased it should be good to go. Although I just started using system versions on CentOS 7 and I haven't yet ran into a problem.

**Comment by adamjstewart** [comment]
Closing as this PR hasn't been updated in over a year and there is no evidence that the system versions of software need to be listed in the package in order to use them as external packages.

**Pull request closed without merging**